File: Love-Babbar-450-In-CSharp/03_string/01-reverse-string.cs
using System;
using System.Collections.Generic;
using System.Text;
using Xunit;

namespace stringg
{/*
    link: https://leetcode.com/problems/reverse-string/
*/

    public class _01_reverse_string
    {
        [Fact]
        public void reverse_StringTest()
        {
            var strArr = "123456".ToCharArray();
           var result =  reverseString(strArr);
            rvereseArrayRecursion(strArr, 0, strArr.Length-1);
        }
        public string reverseString(char[] s)
        {
            int len = s.Length;
            for (int i = 0; i < len / 2; i++)
            {
                char temp = s[i];
                s[i] = s[s.Length - 1-i];
                s[s.Length - 1 - i] = temp;
            }
            return new string(s);
        }
        // TC: O(N)
        private void rvereseArrayRecursion(char[] arr, int start, int end)
        {
            if (start >= end) return;
            var temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            rvereseArrayRecursion(arr, start + 1, end - 1);
        }

    }
}

File: Love-Babbar-450-In-CPPTest/03_string/01-reverse-string.cpp
/*
	link: https://leetcode.com/problems/reverse-string/
*/

#include "CppUnitTest.h"
#include <iostream>
#include <algorithm>
#include<vector>
using namespace std;
using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace LoveBabbar450InCPPTest
{
	TEST_CLASS(StringClass)
	{
	public:

		TEST_METHOD(reverseStringTest)
		{
			vector<char> str = {'1','2'};
			reverseString(str);
		}
		// TC: O(N)
		void rvereseArrayRecursion(char arr[], int start, int end)
		{
			if (start >= end) return;
			auto temp = arr[start];
			arr[start] = arr[end];
			arr[end] = temp;
			rvereseArrayRecursion(arr, start + 1, end - 1);
		}
		void reverseString(vector<char>& s) {
			int len = s.size();
			for (int i = 0; i < len / 2; i++) {
				swap(s[i], s[len - i - 1]);
			}
		}
	};
}
